packages/squiggle-lang/src/public/SqValueContext.ts

```diff
@@ -1,15 +1,10 @@
-import { AST, ASTNode } from "../ast/parse.js";
+import { ASTNode } from "../ast/parse.js";
 import { isBindingStatement } from "../ast/utils.js";
-import { SqProject } from "./SqProject/index.js";
+import { RunContext } from "./SqProject/ProjectItem.js";
 import { SqValuePath, SqValuePathEdge } from "./SqValuePath.js";
 
 export class SqValueContext {
-  public project: SqProject;
-  public sourceId: string;
-  public source: string;
-
-  // top-level AST; we pull comments from it for docstrings
-  public ast: AST;
+  public runContext: RunContext;
```

Comment:
One example of why this change is good is that each `RunContext` points to the correct environment (`SqProject` is mutable!).

In `useSimulator`, we have code that caches the environment when the run starts; it shouldn't be necessary now. I haven't done this cleanup in this PR, though.